chapters/interface.tex

```diff
@@ -594,10 +594,11 @@ \section{Type Compatible Expressions}\label{type-compatible-expressions}
 \item Binary operators if both operands are of simple types, e.g.\ \lstinline!A + B!.
 Binary operators for other types are only defined for operator records, \cref{overloaded-binary-operations}, and do not necessarily require that the operands are type compatible with each other.
 \end{itemize}
-The resulting type of the expression in case of two type compatible subexpressions \lstinline!A! and \lstinline!B! is defined as follows:
+If the subexpressions satisfy the following restrictions they are called type compatible expressions, and the resulting type of the expression is the type compatible expression.
+Otherwise the expression is illegal.
 \begin{itemize}
 \item
-  If \lstinline!A! is a record expression \lstinline!B! must also be a record expression with the same named elements.
+  If \lstinline!A! is a record expression, \lstinline!B! must also be a record expression with the same named elements.
   The type compatible expression is a record comprised of named elements that are compatible with the corresponding named elements of both \lstinline!A! and \lstinline!B!.
```

Comment:
State clearly when we talk about the result:
```suggestion
  The type of the resulting expression is a record comprised of named elements that are type compatible with the corresponding named elements of both \lstinline!A! and \lstinline!B!.
```

Reply:
Ok, added extra "type".